File: src/infrastructure/entities/Tag.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, Unique } from "typeorm";

@Entity("tags")
@Unique(["slug"])
export class Tag {
  @PrimaryGeneratedColumn("increment", { type: "bigint" })
  id: number;

  @Column({ type: "varchar", length: 100, nullable: false })
  name: string;

  @Column({ type: "varchar", length: 100, unique: true, nullable: false })
  slug: string;
}

File: src/main/java/com/example/demo/tag/entity/TagEntity.java
package com.example.demo.tag.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "tags")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class TagEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "name", columnDefinition = "VARCHAR(100)", nullable = false)
    private String name;
    @Column(name = "slug", columnDefinition = "VARCHAR(100)", nullable = false, unique = true)
    private String slug;
}
